WordPress/gutenberg
# Add block class name to the list block

## What?

Adds the `wp-block-list` class to the list block
For https://github.com/WordPress/gutenberg/issues/12420

## Why?

The missing class makes it difficult to style the block separately from other lists.

## How?

List:
- Removes `"className": false,` from block.json.
- Removes the "`__experimentalSelector": "ol,ul",` from block.json so that the styles are applied to `.wp-block-list` instead of plain ul and ol.

List item:
- Changes the selector for the list-item block to `.wp-block-list > li`

## Testing Instructions
Activate a block theme.

Open theme.json in your code editor and add a visible style to the list and list-item blocks, for example a background color.
If you are using Twenty Twenty-Four, the theme already has a styles for the list block, find them on line 515 in theme.json to avoid validation errors.

Example:
```
			"core/list": {
				"color": {
					"background": "pink"
				},
				"spacing": {
					"padding": {
						"left": "var(--wp--preset--spacing--10)"
					}
				}
			},
			"core/list-item":{
				"color": {
					"background": "yellow"
				}
			},
```

Create a new post or page.
Add an HTML block with a list inside.
Example:
```

<ul><li>hello</li><li>world</li></ul>

```
Select preview from the HTML block toolbar and confirm that the list does not have a background color.

Add a list block with some inner list items.
- Confirm that the block background is pink and the list item background is yellow.
- View the page source and confirm that the `wp-list-block` is present on the block.
- Open the code editor mode and confirm that the `wp-list-block` is present on the block.

Save, and view the front of the website.

- Confirm that the list in the HTML block does not have a background color.
- Confirm that the list block has a pink background, and list item has a yellow background.
- View the page source and confirm that the wp-list-block is present on the block.

Return to the editor and style the blocks via the settings in the sidebar; adjust color, typography and spacing. Confirm that the custom styles are applied correctly in the editor and on the front.

Bonus:
If you are using Twenty Twenty-Four, you can also confirm that the navigation block (which uses a list element) in the theme's header and footer does not have a pink background.


## Screenshots or screencast
